### ISSUE TITLE
Train an LSTM to achieve a test-set perplexity of 56.5

### ISSUE THREAD

**Issue by zbh2047** [JingzhaoZhang/why-clipping-accelerates#2]
In the paper it is said that "After training, the model reaches a test-set perplexity of 56.5". Is there some useful tricks to obtain the results? It seems that after running the awd-lstm-lm code using gradient clipping without ASGD, I can only get a perplexity of 75. I think the problem may be caused by:
(1) the lr decay strategy. I tried both constant lr and lr decay (by 10x for 3 times), but the results are all bad.
(2) regularization hyper-parameters. I use the same hyperparameter in the original paper: Regularizing and Optimizing LSTM Language Models.
(3) the optimizer. I only use clipped sgd in the whole training process. I don't use AGSD.

Would you share some details about training a model with a test-set perplexity of 56.5? Thank you very much!

**Comment by zbh2047** [comment]
It seems that switching to Pytorch 0.4 can greatly boost the performance. I will close this issue. Thank you!
